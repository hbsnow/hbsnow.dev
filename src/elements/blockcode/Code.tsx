import React, { PropsWithChildren } from "react";

type Props = Readonly<PropsWithChildren<unknown>>;

const Code = (props: Props): JSX.Element => {
  const { children } = props;

  return (
    <code data-testid="Code" className="code">
      {children}
      <style jsx>{`
        .code {
          display: block;
        }
      `}</style>
    </code>
  );
};

export default Code;
