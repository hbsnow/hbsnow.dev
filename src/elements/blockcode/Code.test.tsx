/* eslint-env jest */
import React from 'react'

import renderer from 'react-test-renderer'

import Code from './Code'

describe(Code.name, () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Code>blockcode</Code>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
