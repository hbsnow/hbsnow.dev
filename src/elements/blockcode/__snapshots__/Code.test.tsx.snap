// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code renders correctly 1`] = `
<code
  className="jsx-4275953942 code"
  data-testid="Code"
>
  blockcode
</code>
`;
